pages/account.tsx

```diff
@@ -0,0 +1,303 @@
+  Box,
+  Button,
+  Flex,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Heading,
+  HStack,
+  Input,
+  Spacer,
+  StackProps,
+  Text,
+  useDisclosure,
+  VStack,
+} from "@chakra-ui/react";
+import ImageUploadModal from "@components/ImageUpload/ImageUploadModal";
+import Layout from "@components/Layout";
+import LoginModal from "@components/LoginModal";
+import { getCurrentUser } from "@lib/helpers/backend/user-helpers";
+import pageTitle from "@lib/helpers/frontend/page-title";
+import { CreateResponseObject } from "@lib/helpers/type-utilities";
+import {
+  useDeleteUserMutation,
+  User,
+  useUpdateUserMutation,
+  useUserQuery,
+} from "@lib/hooks/user";
+import { GetServerSideProps, NextPage } from "next";
+import Head from "next/head";
+import Image from "next/image";
+import { FormEvent, useRef, useState } from "react";
+import { v4 as uuid } from "uuid";
+
+const PAGE_TITLE = pageTitle("Account");
+
+interface AccountPageProps {
+  initialUser: User | null;
+}
+
+const AccountPage: NextPage<AccountPageProps> = ({ initialUser }) => {
+  const { user } = useUserQuery(initialUser ?? undefined);
+
+  if (!user) {
+    return (
+      <>
+        <Head>
+          <title>{PAGE_TITLE}</title>
+        </Head>
+        <Layout>
+          <LoginModal
+            isOpen={true}
+            onClose={() => {}}
+            message="To access the account page you must be signed in. Please login with your MacID below."
+          />
+        </Layout>
+      </>
+    );
+  }
+
+  const PageHeader = (
+    <Flex
+      direction={{ base: "column", md: "row" }}
+      mt="12"
+      mb="6"
+      gap={{ base: "8", md: "16" }}
+      align="start"
+    >
+      <EditImageWidget user={user} display={{ base: "none", md: "initial" }} />
+      <VStack align="start" width="100%">
+        <Heading as="h1" size="lg" fontWeight="semibold" fontFamily="title">
+          Your Account
+        </Heading>
+        <Text color="tertiaryText" fontWeight="semibold">
+          {user.email}
+          <Text
+            as="span"
+            display="block"
+            fontSize="xs"
+            color="captionText"
+            aria-label="User ID"
+          >
+            ({user.id})
+          </Text>
+        </Text>
+
+        <EditImageWidget user={user} display={{ md: "none" }} pt="4" />
+
+        <Flex
+          direction={{ base: "column", md: "row" }}
+          pt="5"
+          align="start"
+          gap={{ base: "6", lg: "12" }}
+          width="100%"
+        >
+          <UpdateNameForm user={user} />
+          <DeleteAccountForm user={user} />
+        </Flex>
+      </VStack>
+    </Flex>
+  );
+
+  return (
+    <>
+      <Head>
+        <title>{PAGE_TITLE}</title>
+      </Head>
+      <Layout extendedHeader={PageHeader}>
+        {/* TODO: Allow editing and deleting of posts */}
+        <Spacer height="lg" />
+      </Layout>
+    </>
+  );
+};
+
+function EditImageWidget({ user, ...props }: { user: User } & StackProps) {
```

Comment:
Tbh I should probably break these up into multiple files since the file is getting pretty long. I was just getting lazy when finishing this at 2AM. There's nothing wrong with this in general though, as long as you don't go overboard with.